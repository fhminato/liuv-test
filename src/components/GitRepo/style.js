import styled from 'styled-components/native';

export const Container = styled.View`
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 10px;
  border: 1px solid #777;
  border-radius: 5px;
  justify-content: space-between;
  background: #fff;
  margin: 10px;
`;

export const FirstInfo = styled.View`
  display: flex;
`;

export const Button = styled.TouchableOpacity`
  background-color: #5000ca;
  justify-content: center;
  align-items: center;
  height: 25px;
  border-radius: 100px;
  padding: 10px;
`;

export const ButtonText = styled.Text`
  color: #fff;
  text-transform: uppercase;
  font-family: 'Antipasto Pro';
  font-size: 11px;
`;

export const Info = styled.Text`
  text-transform: uppercase;
  font-family: 'Antipasto Pro';
`;
