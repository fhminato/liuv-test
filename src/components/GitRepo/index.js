import React from 'react';
import {Container, FirstInfo, Info, Button, ButtonText} from './style';

export default function GitRepo(props) {
  return (
    <Container>
      <FirstInfo>
        <Info>{props.item.name}</Info>
        <Info>{props.item.language}</Info>
      </FirstInfo>
      <Button onPress={() => props.onClick(props.item)}>
        <ButtonText>Detalhes</ButtonText>
      </Button>
    </Container>
  );
}
