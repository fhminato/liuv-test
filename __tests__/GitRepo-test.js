import React from 'react';
import GitRepo from '../src/components/GitRepo';

import renderer from 'react-test-renderer';

const item = {
  name: 'Liuv',
  language: 'React Native',
};

test('renders correctly', () => {
  const tree = renderer
    .create(<GitRepo item={item} onClick={() => console.log()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
