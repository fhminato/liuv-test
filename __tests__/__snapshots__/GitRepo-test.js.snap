// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#777",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Antipasto Pro",
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Liuv
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Antipasto Pro",
            "textTransform": "uppercase",
          },
        ]
      }
    >
      React Native
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#5000ca",
        "borderRadius": 100,
        "height": 25,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Antipasto Pro",
            "fontSize": 11,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Detalhes
    </Text>
  </View>
</View>
`;
